packages/penrose-core/src/compiler/Style.ts

```diff
@@ -361,18 +367,21 @@ const checkRelPattern = (varEnv: Env, rel: RelationPattern): StyErrors => {
 
     // TODO(error)
     if (isErr(res2)) {
-      // TODO(error): extract and return error -- how to convert from SubstanceError to StyError?
-      return ["substance typecheck error in E"];
+      // TODO(error): extract and return error -- how to convert from SubstanceError to StyleError?
+      const subErr2: SubstanceError = unsafelyGetErr(res2);
```

Comment:
Again. you can use `res2.error` here. Looks like you are actually checking Style AST nodes transformed to Substance ones? If so, your source locs will be Style-specific? In general, the `ASTNode` type doesn't have file related information yet. Maybe we'll need it later.

Reply:
>  Looks like you are actually checking Style AST nodes transformed to Substance ones? If so, your source locs will be Style-specific? In general, the ASTNode type doesn't have file related information yet. Maybe we'll need it later.

Yeah, this should return Style source location. Since I'm reporting from Style, I can customize the error (as I guess you were suggesting here? https://github.com/penrose/penrose/pull/473#discussion_r574171134) to specify that the file is Style.

Reply:
Yep, that was my suggestion. If you already have the right locations, we can just report this error as-is for now. File info is probably another PR.